### ISSUE TITLE
Querying based on extended attribute

### ISSUE THREAD

**Issue by dragid10** [kuzmoyev/google-calendar-simple-api#102]
Is it currently possible to at all query based on an extended attribute. Something I'm wanting to do is set the creator field for each event created to my scripts name, and then on subsequent runs be able to find just those events that my script created and delete them in necessary.

If thats not currently available, I can take a look into seeing if I'd be able to add that

**Comment by kuzmoyev** [comment]
I believe, the creator field is read-only, so you can't set it manually

**Comment by kuzmoyev** [comment]
As for searching, you can use [`get_events`](https://google-calendar-simple-api.readthedocs.io/en/latest/code/google_calendar.html#gcsa.google_calendar.GoogleCalendar.get_events) method with `query` parameter.

**Comment by kuzmoyev** [comment]
Ohh. You've meant the creator field in the extended properties. I guess we could implement the querying. Will think about it

**Comment by JulesGrd** [comment]
you can define a custom id which begins with a sequence of your choice and which ends with random letters and numbers.
This way you will be able to check if the id of an event starts with the chosen sequence and to know if this event was created by your script.

**Comment by kuzmoyev** [comment]
@JulesGrd Unfortunately, this doesn't solve the issue, as you'd need to retrieve all the events anyway and filter them out locally.

**Comment by JulesGrd** [comment]
Indeed, this does not prevent the fact of retrieving all the events and filtering them locally. This just allows you to know the events created by your script, to modify them later.

**Comment by kuzmoyev** [comment]
Unfortunately, querying on the extended attributes is not supported by Google Calendar API. Hence this filtering would need to happen on the client-side. I don't think it's very useful at this point to implement it into the library as it can be easily done by the developer. So closing the issue.

If you think it needs more attention, please let me know here.
